sdk/include/opentelemetry/sdk/trace/simple_processor.h

```diff
@@ -57,12 +57,14 @@ class SimpleSpanProcessor : public SpanProcessor
   void Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override
   {
     // We only call shutdown ONCE.
-    if (!shutdown_latch_.test_and_set(std::memory_order_acquire))
+    if (exporter_ != nullptr && !shutdown_latch_.test_and_set(std::memory_order_acquire))
```

Comment:
Currently we implicitly assume that `exporter_` is not `nullptr`. Maybe we should make that explicit in the doc string, instead of adding `nullptr` checks all around?

Because, to be thorough, we'd also to need all `nullptr` checks on the `Export` call in `OnEnd`, and around the `MakeRecordable` call.

Reply:
Looked at this a little more and I think it is fine to only check `exporter_ != nullptr` in `Shutdown` call, because `Shutdown` is supposed to be called in destructor, and the span processor could be created like `new SimpleSpanProcessor(nullptr)`, we'd better destruct this object gracefully. For other function like `Export`, `OnEnd`, missing `exporter_` might mean an issue in the user code, so we could trigger and throw nullptr access exception.

Besides, `Shutdown` is declared as `noexcept` but `Export()` is not so it looks fine to throw exception there.